### ISSUE TITLE
[Suggestion] Some way of "extracting" binaries of correct framework to a bin folder

### ISSUE THREAD

**Issue by Vidarls** [fsprojects/Paket#707]
**Tl;Dr;**

I'm suggestion a Paket command to extract binaries, resources and tools from a set of packages into a deployable bin folder. Purpose is to compose a lot of packages into something that can be deployed without using visual studio.

**Background:**

I'm currently working on improving the build automation of our desktop application that is currently undergoing a move from VB6 to .net.
In that process I now have to manage a lot more dependencies of the VB6 project. This was previously horrible, but now with Paket is is quite a bit less horrible. 

My Vb6 build process is now like this:
1. Fetch all .net dependencies using Paket
2. Copy all the relevant dlls (and other supporting stuff like icon resources, translations and such) from the packages to a deployable structure in my bin folder
3. Register all direct VB6 dependencies using Regasm
4. Execute vb6 from command line to build all vb6 projects
5. Create and embed side-by-side interopmanifests for .net assemblies and VB6 applications.
6. DONE binaries are now ready for deploy

Currently I have good tooling through the F# build and dependency management eco-system (FAKE + Paket) for everything except item 2.

**Possible draft for solution:**

I have some code that caters for my specific needs right now: (See this gist: https://gist.github.com/Vidarls/549ce388ea6d219ed417)

The code is in the form of a FAKE build script.

The code is quite general, but I know it fails to handle all the different combinations of target framework sub folders, (and probably has other potentials for being more robust too.)

**Comment by forki** [comment]
I think this is related to https://github.com/fsprojects/Paket/issues/677

**Comment by Vidarls** [comment]
There certainly is some overlap with regards to figure out which binaries to resolve.

**Comment by matthid** [comment]
I think this is "easily" achieved by directly using Paket API (Paket.Core nuget package) and using the InstallModel APIs. Feel free to disagree (in which case I could re-open)

**Comment by forki** [comment]
Nonono. Please don't use paket api.

Am 27.05.2017 1:48 nachm. schrieb "Matthias Dittrich" <
notifications@github.com>:

**Comment by matthid** [comment]
What?

**Comment by forki** [comment]
Fable should just read files in simple format. No need to depend on paket.core. Let's just make it easy for fable

**Comment by matthid** [comment]
Lets continue on https://github.com/fsprojects/Paket/issues/2373
But to answer: Why?
Note that I don't think its a good idea to list references, I'd really prefer fable using paket API instead.
For the group info I don't mind too much and we can add that.

**Comment by forki** [comment]
Yes I responded to wrong issue. Sorry
